From f64c88e36b304c189ffb31eb98fddec97c974d20 Mon Sep 17 00:00:00 2001
From: Martin Cavoj <martin.cavoj@renuo.ch>
Date: Fri, 24 May 2019 13:42:51 +0200
Subject: [PATCH] Patch events

---
 events.js | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/events.js b/events.js
index 1391f52..849e9a8 100644
--- a/events.js
+++ b/events.js
@@ -20,10 +20,15 @@ function listen(res, group, groupEvent, handler) {
   res.on('finish', removeListener);
 }
 
+function corsSse(_req, resp, next) {
+  resp.setHeader('Access-Control-Allow-Origin', '*')
+  next()
+}
+
 module.exports = group => {
   const router = express.Router();
 
-  router.get('/', sse, (req, res) => {
+  router.get('/', corsSse, sse, (req, res) => {
     // Handler
     function sendState() {
       res.json(group.list());
@@ -36,7 +41,7 @@ module.exports = group => {
     listen(res, group, 'change', sendState);
   });
 
-  router.get('/output', sse, (req, res) => {
+  router.get('/output', corsSse, sse, (req, res) => {
     function sendOutput(id, data) {
       res.json({
         id,
-- 
2.21.0

